src/CommandingOfficers/CommanderAbility.java

```diff
@@ -9,14 +9,16 @@
   public static final int PHASE_TURN_END = PHASE_BUY << 1;
   public final Commander myCommander;
   protected String myName;
-  private int myPowerCost;
+  public final double numDiscretes; // Ability cost in a chunky form. Can serve as a reference to the "original cost" of the ability.
```

Comment:
"a chunky form"? I have no clue what this is supposed to mean.

Reply:
Also, if it's supposed to be the original cost of the ability, `numDiscretes` is a terrible name.

Reply:
Also, it doesn't appear to ever be referenced?

Reply:
Two different game modes where you add free power, but in different increments? Why not just have a "free power [amt]" game mode then? There is no reason to do it two ways.
I don't like `discretes` is a noun here - I get why, but this is not ideal. We also add at minimum one power unit each time the ability is used, but I guess you are trying to change that here too, by keeping the same number of units, but each unit is "bigger". Which means the "discretes->funds" problem you describe is self-inflicted by this pull request.

> This uh, wasn't super-thoroughly thought out. 

Yes.

> But it was bugging me

What, exactly? Just that the power costs are not double-precision? That is easily changed without doing the rest of this stuff.

Reply:
> a "free power [amt]" game mode

See, that's where things get a little awkward. How do you define [amt]?
My initial concept had been you get a certain number *of stars*. So you can keep using powers over and over again with little slowing, but the contributions made by combat will become less relevant over time as the costs expand.
It's equally valid to make a game mode that gives you a free charge amount of *funds*, but that's a very different game mode, even with similar amounts, because the stars will scale up with your power costs but the funds will not.

>I don't like discretes is a noun here - I get why, but this is not ideal.

Do you have a better term?

> We also add at minimum one power unit each time the ability is used, but I guess you are trying to change that here too, by keeping the same number of units, but each unit is "bigger". Which means the "discretes->funds" problem you describe is self-inflicted by this pull request.

I didn't like the minimum one ever since I realized it was a whole quanta. The PR was merged by that time, so I couldn't really complain at that point....
It is. It's the result of trying to understand power charge to be two different things at the same time. The carts don't have this problem, because they scale power costs together... but I don't think I want to make that change for the main branch since I think it encourages you to make the best use of *all* tools in your kit, not just the "most cost effective" one.

> What, exactly? Just that the power costs are not floating-point numbers? That is easily changed without doing the rest of this stuff.

I want to be able to treat powers as costing a certain number of quanta, but I also want to be able to scale their costs independently.
Having power charge stored in quanta rather than funds makes treating the power as if it costs only a certain number of quanta rather silly, since it's two copies of the same unit.

I didn't do a terribly good job of communicating my intent, though.

Reply:
> See, that's where things get a little awkward. How do you define [amt]

We do assign the abilities a cost - and they'd build intuition that a cost of 3 generally takes x time to charge, etc. We could also just have [amt] be low, medium, or high, and not say exactly how much they are. Or give a range of options over a reasonable span [0-5] or whatever, and the users would figure it out. While these may not be super intuitive, I don't think a funds-based approach (with a scaling factor of 9000) is going to be any more approachable.

> combat will become less relevant over time as the costs expand

This sounds... kinda antithetical to the whole premise of the game.

> I didn't like the minimum one ever since I realized it was a whole quanta

Fair. I'm inclined to agree - this makes the scaling of low-cost abilities much higher than the scaling of more expensive ones. I like the idea of changing ability cost to a double, which would also encourage treating it as a non-discretized thing, since it could get more expensive by 1.34 or whatever.

> the result of trying to understand power charge to be two different things at the same time

An argument for not doing that, I think. Plus drawing things might get strange if your small "stars" are all of a sudden more expensive than your large ones.